tannerlinsley/react-query
# asyncStoragePersistor for React Native

**Implementation**

I looked at the createLocalStoragePersistor and just changed a few lines and necessary imports. Please feel free to suggest if there is anything I should change or improve.

**How to use**

```
import {asyncStoragePersistor} from './utility/createAsyncStoragePersistor';

const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      cacheTime: 1000 * 60 * 60 * 48, // 48 hours
      staleTime: 1000 * 60 * 60 * 24, // 24 hours
    },
  },
});

const storagePersistor = asyncStoragePersistor({
  asyncStorageKey: 'AsyncStorageReactQuery', // asyncStorageKey vs localStorageKey
  throttleTime: 1000,
});

persistQueryClient({
  queryClient,
  persistor: storagePersistor,
  maxAge: 1000 * 60 * 60 * 24, // 24 hours in milliseconds
});
```

**How to test**

Use some queries to get the data. Restart the React Native app. We should see the UIs with previously called queries immediately without needing to call the APIs.
